### PR TITLE
Upgrade to Realm Core v11.12.0

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

Realm Core v11.12.0 unifies the CMake files for Apple's operating systems which means we can simplify our set up.

## ☑️ ToDos

* [x] 📝 Changelog entry
* ~[ ] 📝 `Compatibility` label is updated or copied from previous entry~
* ~[ ] 🚦 Tests~
* ~[ ] 📱 Check the React Native/other sample apps work if necessary~
* ~[ ] 📝 Public documentation PR created or is not necessary~
* ~[ ] 💥 `Breaking` label has been applied or is not necessary~

*If this PR adds or changes public API's:*
* ~[ ] typescript definitions file is updated~
* ~[ ] jsdoc files updated~
* ~[ ] Chrome debug API is updated if API is available on React Native~

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,9 @@ x.x.x Release notes (yyyy-MM-dd)
 * Converting floats/doubles into Decimal128 would yield imprecise results. ([realm/realm-core#5184](https://github.com/realm/realm-core/pull/5184), since v6.1.3)
 * Using accented characters in class and property names may end the sync session ([realm/realm-core#5196](https://github.com/realm/realm-core/pull/5196), since v10.3.0-rc.1)
 * Waiting for upload after opening a bundled Realm file could hang. ([realm/realm-core#5277](https://github.com/realm/realm-core/issues/5277), since v10.10.0)
+* Realm Query Language would not accept `in` as a property name. ([realm/realm-core#5312](https://github.com/realm/realm-core/issues/5312))
+* Fixed an issue that could lead to a crash with exceptions like `'KeyNotFound'`. ([realm/realm-core#5283](https://github.com/realm/realm-core/issues/5283), since v6.0.0)
+* Refreshing the user profile after the app has been destroyed leads to a failure. ([realm/realm-dotnet#2800](https://github.com/realm/realm-dotnet/issues/2800)
```
Comment (takameyer) [inline CHANGELOG.md:17]: Should we link to the core issue instead?
https://github.com/realm/realm-core/issues/5238

File: CMakeLists.txt [hunk 2]
```diff
@@ -15,8 +15,8 @@ if(DEFINED CMAKE_JS_VERSION)
             set(CMAKE_TOOLCHAIN_FILE "C:\\src\\vcpkg\\scripts\\buildsystems\\vcpkg.cmake")
         endif()
     elseif(APPLE)
-        set(CMAKE_TOOLCHAIN_FILE "${CMAKE_CURRENT_LIST_DIR}/vendor/realm-core/tools/cmake/macosx.toolchain.cmake")
-        if(DEFINED ENV{NODE_ARCH_ARM})
+      set(CMAKE_OSX_DEPLOYMENT_TARGET "10.9")
```
Comment (takameyer) [inline CMakeLists.txt:18]: Why was this change made?

**Review by takameyer — APPROVED** [review]
Just a couple questions.  I approve!

File: CHANGELOG.md (hunk 1 above)
Comment (kneth) [reply CHANGELOG.md:17]: I am trying to link to the issue reported by a user

File: CMakeLists.txt (hunk 2 above)
Comment (kneth) [reply CMakeLists.txt:18]: https://github.com/realm/realm-core/pull/5271

**Pull request merged**